### PR TITLE
#55 Added tower enemy detection

### PR DESCRIPTION
**Pull request by Andrea1141**
Before submitting a Pull Request please read and tick this checkbox:

- [x] I did read the [project licensing](https://github.com/crystal-bit/hacktoberfest-2020#license) and I agree with the content of this Pull Request being licensed under the same conditions.

I added enemy detection to the tower and I changed the functions' order following code style in readme.md

### CODE REVIEW

File: scenes/tower/tower.gd [hunk 1]
```diff
@@ -35,35 +39,61 @@ func _ready() -> void:
...
+	update()
+
+
+func _on_AttackRangeShowArea_mouse_exited() -> void:
+	alpha_area = alpha_area_hide
+	update()
+
+func _on_Area2D_body_entered(body):
+	if body is Enemy:
+		change_state(2)
```
Comment (davcri) [inline scenes/tower/tower.gd:62]: Can you replace the integers with the enum key? Like here (and in next occurrences) it's possible to use `TOWER_STATES.ATTACKING`

**Review by davcri — CHANGES_REQUESTED** [review]

**Comment by davcri** [comment]
Hi @Andrea1141 I requested a small change in the code.

Also it would be nice to have to tower preserve the current enemy target.  
Currently I think that the tower switches to a new target enemy each time a new enemy body enters the area.

Also there could be another possible issue, but we will take care of this in the future if needed.

File: scenes/tower/tower.gd (hunk 1 above)
Comment (Andrea1141) [reply scenes/tower/tower.gd]: Solved in d4f6d47!

**Review by davcri — APPROVED** [review]

**Comment by davcri** [comment]
@Andrea1141 thank you very much for updating the pull request so quickly! Now the PR is perfect :) I'm merging it right now!

**Pull request merged**